sqlmesh/core/engine_adapter/snowflake.py

```diff
@@ -99,6 +101,97 @@ def snowpark(self) -> t.Optional[SnowparkSession]:
             ).getOrCreate()
         return None
 
+    def create_engine_managed_table(
+        self,
+        table_name: TableName,
+        query: Query,
+        columns_to_types: t.Optional[t.Dict[str, exp.DataType]] = None,
+        partitioned_by: t.Optional[t.List[exp.Expression]] = None,
+        clustered_by: t.Optional[t.List[str]] = None,
+        table_properties: t.Optional[t.Dict[str, exp.Expression]] = None,
+        table_description: t.Optional[str] = None,
+        column_descriptions: t.Optional[t.Dict[str, str]] = None,
+        **kwargs: t.Any,
+    ) -> None:
+        target_table = exp.to_table(table_name)
+
+        # Snowflake defaults to uppercase and it also makes the property presence checks
+        # easier
+        table_properties = {k.upper(): v for k, v in (table_properties or {}).items()}
+
+        # the WAREHOUSE property is required for a Dynamic Table
+        if "WAREHOUSE" not in table_properties:
+            table_properties["WAREHOUSE"] = self.current_warehouse
+
+        # so is TARGET_LAG
+        # ref: https://docs.snowflake.com/en/sql-reference/sql/create-dynamic-table
+        if "TARGET_LAG" not in table_properties:
+            raise SQLMeshError(
```

Comment:
Can we also validate this in the model definition to make the feedback loop faster.

Reply:
I guess we could for `TARGET_LAG` but not for `WAREHOUSE` since that requires access to either the DB connection config or the ability to run queries

Reply:
Done, i've added a branch in `validate_definition` although this method in looks like it might need to be refactored at some point because it does a bunch of checks that perhaps should live in the model Kind classes